src/k8s-extension/azext_k8s_extension/partner_extensions/ContainerInsights.py

```diff
@@ -244,7 +245,10 @@ def _ensure_default_log_analytics_workspace_for_monitoring(cmd, subscription_id,
     cluster_resource_id = '/subscriptions/{0}/resourceGroups/{1}/providers/{2}/{3}/{4}'.format(
         subscription_id, cluster_resource_group_name, cluster_rp, cluster_type, cluster_name)
     try:
-        resource = resources.get_by_id(cluster_resource_id, '2022-05-01-preview')
+        if cluster_rp.lower() == consts.HYBRIDCONTAINERSERVICE_RP:
+            resource = resources.get_by_id(cluster_resource_id, '2022-05-01-preview')
+        else:
+            resource = resources.get_by_id(cluster_resource_id, '2020-01-01-preview')

```

Comment:
can you please use the value stored in constants file - consts.CONNECTED_CLUSTER_API_VERSION

Reply:
this version I discussed with other teammates, since we are not sure whether there might be other rp type, for now just revert to the previous working version